### PR TITLE
Introduce specific annotations for the generators

### PR DESCRIPTION
**Pull request by andreaTP**
## Description
Follow up from #4298 , introducing annotations for the fields that we want to support/map.

## Type of change
 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [X] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [ ] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [X] Code contributed by me aligns with current project license: [Apache 2.0](https://www.apache.org/licenses/LICENSE-2.0)
 - [ ] I Added [CHANGELOG](https://github.com/fabric8io/kubernetes-client/blob/master/CHANGELOG.md) entry regarding this change
 - [X] I have implemented unit tests to cover my changes
 - [ ] I have added/updated the [javadocs](https://www.javadoc.io/doc/io.fabric8/kubernetes-client/latest/index.html) and other [documentation](https://github.com/fabric8io/kubernetes-client/blob/master/doc/CHEATSHEET.md) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=fabric8io_kubernetes-client) report
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

File: generator-annotations/src/main/java/io/fabric8/generator/annotation/Max.java
Comment (manusa) [inline generator-annotations/src/main/java/io/fabric8/generator/annotation/Max.java:24]: For the follow-ups, it might be interesting to add Javadoc to each annotation properly documenting its usage, and maybe linking the Kubernetes spec equivalent.

**Review by manusa — APPROVED** [review]

File: generator-annotations/src/main/java/io/fabric8/generator/annotation/Max.java
Comment (andreaTP) [reply generator-annotations/src/main/java/io/fabric8/generator/annotation/Max.java:24]: I'm failing to find a good "linkable" source with the specific field description, e.g.:
https://kubernetes.io/docs/reference/kubernetes-api/extend-resources/custom-resource-definition-v1/

Do you have any idea?

File: crd-generator/api/src/main/java/io/fabric8/crd/generator/AbstractJsonSchema.java
Comment (metacosm) [inline crd-generator/api/src/main/java/io/fabric8/crd/generator/AbstractJsonSchema.java:82]: All annotations should be put in the same package, imo.

Comment (andreaTP) [reply crd-generator/api/src/main/java/io/fabric8/crd/generator/AbstractJsonSchema.java:82]: At this point we have annotations in 2 places:
 - `io.fabric8.generator.annotation` -> generic modifiers to represent CRD properties in Java code
 - `io.fabric8.crd.generator.annotation` -> specific annotations to tweak the specifically the `crd-generator`
 
 Do you see any other invariant @metacosm ?

File: generator-annotations/src/main/java/io/fabric8/generator/annotation/Max.java
Comment (manusa) [reply generator-annotations/src/main/java/io/fabric8/generator/annotation/Max.java:24]: No, I only found the go types sources when I was evaluating the field precision of the Max and Min configurations.

File: crd-generator/api/src/main/java/io/fabric8/crd/generator/AbstractJsonSchema.java
Comment (metacosm) [reply crd-generator/api/src/main/java/io/fabric8/crd/generator/AbstractJsonSchema.java:82]: As soon as the annotations are used in different contexts, it would be better for users to help discoverability, in particular, for the annotations to be in the same location even though they initially were developed for different purposes… Obviously, changing annotations location at this point wouldn't be backwards compatible so maybe just a matter of documentation? It might make sense to explain why they live in different packages so that users (and future maintainers, down the line 😸) don't think it's an arbitrary decision.

Comment (andreaTP) [reply crd-generator/api/src/main/java/io/fabric8/crd/generator/AbstractJsonSchema.java:82]: Added a brief JavaDoc on each annotation, let me know if you want other/more changes for this 🙂

File: generator-annotations/src/main/java/io/fabric8/generator/annotation/Max.java
Comment (andreaTP) [reply generator-annotations/src/main/java/io/fabric8/generator/annotation/Max.java:28]: added a generic reference to: `jsonschemaprops`  K8s API reference ... the best I have found in the wild ...

**Review by metacosm — APPROVED** [review]

**Comment by andreaTP** [comment]
@manusa ready to merge?

**Comment by manusa** [comment]
Releasing 6.1.1 ATM 
Once released and checked I'll start merging the pending PRs

**Review by manusa — APPROVED** [review]

**Comment by andreaTP** [comment]
Thanks for the heads up @manusa !

**Comment by manusa** [comment]
> WDYT? Should we move on with a separate java-generator issue?

I'm merging this one, so please, do so.

**Pull request merged**
